packages/gasket-intl/lib/internal-intl-manager.js

```diff
@@ -0,0 +1,140 @@
+import { isBrowser, LocaleFileStatus } from './constants.js';
+import { LocaleHandler } from './locale-handler.js';
+
+/** @type {import('./index.d.ts').LocaleHandler} */
+let _browserSingletonHandler;
+
+/**
+ * Utility class for loading locale files
+ * @type {import('./internal.d.ts').InternalIntlManager}
+ */
+export class InternalIntlManager {
+  /** @type {import('./internal.d.ts').MessagesRegister } */
+  messagesRegister = {};
+  /** @type {import('./internal.d.ts').StatusRegister } */
+  statusRegister = {};
+  /** @type {import('./internal.d.ts').PromisesRegister } */
+  promisesRegister = {};
+
+  /** @type {import('./internal.d.ts').IntlManager_constructor } */
+  constructor(manifest) {
+    this.manifest = manifest;
+    this.managedLocales = [...this.locales, ...Object.keys(this.manifest.localesMap ?? {})];
+
+    this.init();
+  }
+
+  get locales() {
+    return this.manifest.locales;
+  }
+
+  get defaultLocaleFilePath() {
+    return this.manifest.defaultLocaleFilePath;
+  }
+
+  get staticLocaleFilePaths() {
+    return this.manifest.staticLocaleFilePaths ?? [];
+  }
+
+  /** @type {import('./internal.d.ts').IntlManager_resolveLocale } */
+  resolveLocale(locale) {
+    const { defaultLocale, locales, localesMap = {} } = this.manifest;
+
+    if (locale in localesMap) {
+      return localesMap[locale];
+    }
+
+    if (locales.includes(locale)) {
+      return locale;
+    }
+
+    // attempt fallback to language
+    if (locale.indexOf('-') > 0) {
+      return this.resolveLocale(locale.split('-')[0]);
+    }
+
+    return defaultLocale;
+  }
+
+  /** @type {import('./internal.d.ts').IntlManager_init } */
+  init() {
+    if (isBrowser) {
+      const content = (document.getElementById('GasketIntl') ?? {}).textContent;
```

Comment:
```suggestion
      const element = document.getElementById('GasketIntl');
      const content = element ? element.textContent : '';
```
Is this easier to read?